packages/shell-api/src/helpers.spec.ts

```diff
@@ -23,6 +23,54 @@ describe('dataFormat', () => {
   });
 });
 
+describe('assertArgsDefinedType', () => {
+  it('allows to specify an argument must be defined', () => {
+    try {
+      assertArgsDefinedType([1, undefined], [true, true], 'helper.test');
+    } catch (e) {
+      expect(e.message).to.contain('Missing required argument at position 1');
+      expect(e.message).to.contain('helper.test');
+      return;
+    }
+    expect.fail('Expected error');
+  });
+  it('allows to specify a single argument type', () => {
+    [null, 2, {}].forEach(value => {
+      try {
+        assertArgsDefinedType([1, value], [true, 'string'], 'helper.test');
+      } catch (e) {
+        expect(e.message).to.contain('Argument at position 1 must be of type string');
+        expect(e.message).to.contain('helper.test');
+        return;
+      }
+      expect.fail('Expected error');
+    });
+    expect(() => assertArgsDefinedType([1, 'test'], [true, 'string'])).to.not.throw;
+  });
+  it('allows to specify multiple argument types', () => {
+    [null, {}].forEach(value => {
+      try {
+        assertArgsDefinedType([1, value], [true, ['number', 'string']]);
+      } catch (e) {
+        return expect(e.message).to.contain('Argument at position 1 must be of type number or string');
+      }
+      expect.fail('Expected error');
+    });
+    expect(() => assertArgsDefinedType([1, 'test'], [true, ['number', 'string']])).to.not.throw;
+    expect(() => assertArgsDefinedType([1, 2], [true, ['number', 'string']])).to.not.throw;
+  });
+  it('allows to specify an optional argument with type', () => {
+    expect(() => assertArgsDefinedType([1, undefined], [true, [undefined, 'string']])).to.not.throw;
+    expect(() => assertArgsDefinedType([1, 'test'], [true, [undefined, 'string']])).to.not.throw;
+    try {
+      assertArgsDefinedType([1, 2], [true, [undefined, 'string']]);
+    } catch (e) {
+      return expect(e.message).to.contain('Argument at position 1 must be of type string');
+    }
+    expect.fail('Expected error');
+  });
+});
```

Comment:
Thanks for adding those, so easy to _assert_ how this new method works at a glance now!